probe-rs/src/bin/probe-rs/cmd/debug.rs

```diff
@@ -355,53 +384,53 @@ impl DebugCli {
                     1
                 };
 
-                let mut buff = vec![0u32; num_words];
+                let mut buff = vec![0u16; num_words];
 
                 if num_words > 1 {
-                    cli_data.core.read_32(address, &mut buff)?;
+                    cli_data.core.read_16(address, &mut buff)?;
                 } else {
-                    buff[0] = cli_data.core.read_word_32(address)?;
+                    buff[0] = cli_data.core.read_word_16(address)?;
                 }
 
                 for (offset, word) in buff.iter().enumerate() {
-                    println!("0x{:08x} = 0x{:08x}", address + (offset * 4) as u64, word);
+                    println!("0x{:08x} = 0x{:04x}", address + (offset * 2) as u64, word);
                 }
 
                 Ok(CliState::Continue)
             },
         });
 
         cli.add_command(Command {
-            name: "read_byte",
-            help_text: "Read 8bit value from memory",
+            name: "read32",
```

Comment:
Can we deduplicate this code while we are at it?

Reply:
how'd you do it? they have to call different functions and use different formatters. Could be done with closures but I feel like it'd end up being more complicated than duplicating code.

Reply:
We are already using closures, so I thought maybe we can easily do it, but maybe not :/